doc/secure.md

```diff
@@ -0,0 +1,88 @@
+
+`assumevalid=1` skips validation for scripts. A all scripts in transactions included in blocks that are ancestors of the _assumevalid_ block are assumed to be valid. While this does introduce a little bit of trust, people can independently check that block is part of the main chain.
+
+Known incidents when BGP hijacking was used to exploit Bitcoin users: https://www.wired.com/2014/08/isp-bitcoin-theft/
+
+Do not use blocks and chainstate from unknown sources. Attackers could be providing you a false blockchain, i.e. one that is valid but not the blockchain that all other nodes are following.
+
+### RPC
+
+The JSON-RPC interface is intended for local access. By default it will only accept connections from localhost, but it can be configured to be accessed for a wider netmask; e.g. a trusted LAN network. Bitcoin Core does not allow anyone to connect to the rpcport. You will need to explicitly allow an IP address to connect to it by using the `rpcallowip=<ip>` option in the _bitcoin.conf_ file. If you set it to 0.0.0.0, it will be open to all IP addresses, but this is not recommended as it is not secure.
+
+If you have a need to access it over an untrusted network like the internet, tunnel it through technology specifically designed for that, such as stunnel (SSL) or a VPN.
+
+[Executable, network access, authentication and string handling](/json-rpc-interface.md#security)
+
+[Multi user systems](https://medium.com/@lukedashjr/cve-2018-20587-advisory-and-full-disclosure-a3105551e78b)
+
+[#12763](https://github.com/bitcoin/bitcoin/pull/12763) added the RPC whitelisting feature that helps enforce application policies for services being built on top of Bitcoin Core. It suggested to not connect your Bitcoin node to arbitrary services, reduce risk and prevent unintended access.
+
+### Wallet
+
+[Encrypting the Wallet](/managing-wallets.md#encrypting-the-wallet)
+
+[Backing Up the Wallet](/managing-wallets.md#backing-up-the-wallet)
+
+Do not buy _wallet.dat_ files from anywhere. There are several websites that sell such files to newbies. You can read more about the issue here: https://github.com/bitcoin-core/gui/issues/77#issuecomment-721927922
+
+Understanding the difference between legacy and descriptor wallets:
+
+There is no XPUB in legacy wallets, `dumpwallet` will return private keys. `listdescriptors` can be used to get XPUB and XPRV for descriptor wallets.
+
+
+### Other
+
+*notify config and commandline options accept shell commands to be based on different events in Bitcoin Core. Be careful while using these options and ensure that the commands are safe either by manual inspection or restrictions/alerts in the system used.
+
+## Developers
+
+ℹ️ It’s a myth that those break cannot build.
+
+- [Developer Notes](/developer-notes.md)
+
+### Secure Programming Practices in C++
```

Comment:
I would move this to a dedicated section in the developer notes and link straight to that. It prevents duplication and also makes this document less intimidating for users.

Reply:
Moved to developer notes in https://github.com/bitcoin/bitcoin/commit/05511d08be7e306a60c758ee2a64edc0cd63478e